tests/validator/test_validation_graph.py

```diff
@@ -178,10 +178,209 @@ def test_ExpectationValidationGraph_get_exception_info(
         right.id: {"exception_info": {right_exception}},
     }
 
-    expectation_validation_graph.update(validation_graph_with_single_edge)
-    exception_info = expectation_validation_graph.get_exception_info(
+    expect_column_values_to_be_unique_expectation_validation_graph.update(
+        validation_graph_with_single_edge
+    )
+    exception_info = expect_column_values_to_be_unique_expectation_validation_graph.get_exception_info(
         metric_info=metric_info
     )
 
     assert left_exception in exception_info
     assert right_exception in exception_info
+
+
+@pytest.mark.unit
+def test_parse_validation_graph(
+    expect_column_value_z_scores_to_be_less_than_expectation_validation_graph: ValidationGraph,
+):
+    available_metrics: Dict[Tuple[str, str, str], Any]
+
+    # Parse input "ValidationGraph" object and confirm the numbers of ready and still needed metrics.
+    available_metrics = {}
+    (
+        ready_metrics,
+        needed_metrics,
+    ) = expect_column_value_z_scores_to_be_less_than_expectation_validation_graph._parse(
+        metrics=available_metrics
+    )
+    assert len(ready_metrics) == 2 and len(needed_metrics) == 9
+
+    # Show that including "nonexistent" metric in dictionary of resolved metrics does not increase ready_metrics count.
+    available_metrics = {("nonexistent", "nonexistent", "nonexistent"): "NONE"}
+    (
+        ready_metrics,
+        needed_metrics,
+    ) = expect_column_value_z_scores_to_be_less_than_expectation_validation_graph._parse(
+        metrics=available_metrics
+    )
+    assert len(ready_metrics) == 2 and len(needed_metrics) == 9
+
+
+@pytest.mark.unit
+def test_populate_dependencies(
+    expect_column_value_z_scores_to_be_less_than_expectation_validation_graph: ValidationGraph,
+):
+    assert (
+        len(
+            expect_column_value_z_scores_to_be_less_than_expectation_validation_graph.edges
+        )
+        == 33
+    )
+
+
+@pytest.mark.unit
+def test_populate_dependencies_with_incorrect_metric_name():
+    class PandasExecutionEngineStub:
+        pass
+
+    PandasExecutionEngineStub.__name__ = "PandasExecutionEngine"
+    pandas_execution_engine_stub = cast(ExecutionEngine, PandasExecutionEngineStub())
+
+    graph = ValidationGraph(execution_engine=pandas_execution_engine_stub)
+
+    with pytest.raises(ge_exceptions.MetricProviderError) as e:
+        graph.build_metric_dependency_graph(
+            metric_configuration=MetricConfiguration(
+                "column_values.not_a_metric", IDDict()
+            ),
+        )
+
+    assert (
+        e.value.message
+        == "No provider found for column_values.not_a_metric using PandasExecutionEngine"
+    )
+
+
+@pytest.mark.integration
+def test_resolve_validation_graph_with_bad_config_catch_exceptions_true():
```

Comment:
I think the code paths here can be tested without an integration test; we're not trying to test that the ExecutionEngine raises errors, we are trying to test that the ValidationGraph properly handles those errors.

This would be a good case for a Mock

Reply:
@donaldheppner Investigating....

Reply:
@donaldheppner This has been done.  Thanks.